lib/internal/dns/promises.js

```diff
@@ -180,8 +180,9 @@ function createResolverPromise(resolver, bindingName, hostname, ttl) {
     req.resolve = resolve;
     req.reject = reject;
     req.ttl = ttl;
+    req.search = isSearch;
 
-    const err = resolver._handle[bindingName](req, hostname);
+    const err = resolver._handle[bindingName](req, hostname, isSearch);
```

Comment:
I may be misunderstanding something, but is it needed to pass an argument `isSearch` while it is already in `req` object and it can be extracted from there?

Reply:
I'm not sure either, how JSland calls into C++ land, but looking at [this block of code](https://github.com/nodejs/node/blob/master/src/cares_wrap.cc#L1806-L1812), I figured this is how I'd have to add an additional parameter, and it worked.

If you can show me an example of unwrapping properties on an object, in C++ land, I'm willing to test this approach out and update if needed.

Reply:
Besides, what would be the better approach? Unwrapping properties would implicitly assume that the property names are part of the API contract, whereas passing in parameters would make the signature explicit.

Reply:
Since on the c++ you already have an `Local<Object>` I think you can just 
```c++
Local<Value> search = req_wrap_obj->Get(env->isolate(), FIXED_ONE_BYTE_STRING(env->isolate(), "search"));
CHECK(search->IsBoolean());
bool is_search = (search.As<Boolean>)->Value();
```

Though, yeah, I'm not sure if this is a correct way of doing things (no need to change this yet). I just thought I'd mention it since it is both in the object and as an argument. Is it needed to set it in the `req` object at all (maybe just an argument will be enough)?

Reply:
not really needed anywhere else, I guess. Will fix this.

Reply:
You need to be consistent, it's not done on the non-promises version. I think the only reason `ttl` is set on `req` is so that we can tack `ttl` on to the response object in `onresolve()`, i.e. to connect the input with the output processing rather than interacting with c-ares at all. So I think it'd be appropriate to not attach anything new to `req` since we don't need it after the query is sent.